classes/modules/target-rule/class-bsf-sb-target-rules-fields.php

```diff
@@ -681,11 +681,29 @@ public static function generate_target_rule_selector( $type, $selection_options,
 
 						// taxonomy options.
 						if ( strpos( $sel_value, 'tax-' ) !== false ) {
-							$tax_id        = (int) str_replace( 'tax-', '', $sel_value );
-							$term          = get_term( $tax_id );
+						
+							$suffix = "";
+						
+							$term_suffix = "";
+						
+							if (preg_match("/\d+(-single-.+)$/", $sel_value, $m)) {
+						
+								$suffix = $m[1];
+						
+								$term_suffix = " (Singulars)";
+						
+							}
+						
+							$tax_id = (int) str_replace( 'tax-', '', $sel_value );
+						
+							$term = get_term( $tax_id );
+						
 							$term_taxonomy = ucfirst( str_replace( '_', ' ', $term->taxonomy ) );
-							$output       .= '<option value="tax-' . $tax_id . '" selected="selected" >' . $term->name . ' - ' . $term_taxonomy . '</option>';
-
+						
+							$output .= '<option value="tax-' . $tax_id . $suffix . '" selected="selected" >' . $term->name . ' - ' . $term_taxonomy . $term_suffix . '</option>';
```

Comment:
As we are changing value the selected will work in the case on users end with their old saved value??

Reply:
@imnavanath No! modification which I did here will not update or affected the selected values for existing user save values; they need to update their save values to reflect the changes.

Reply:
If suppose, I have selected a value in master version and when I'll switch to this update the select will come with my last selected value only or deselect my previous value? Also check & resolve codesniffer failed GitHub pipeline.

Reply:
Yes @imnavanath change come with latest selected value for this perticluar Singulars case, & rest selecetd sidebar/values will remain same .

Reply:
Resolved odesniffer failed GitHub pipeline.